### PR TITLE
Update Laplacian jump indicator tests

### PR DESCRIPTION
**Pull request by jwpeterson**
These tests were previously solving an incorrect weak form of the advection-diffusion equation that somehow "worked" although @roystgnr and myself remain stumped as to why. This PR adds test Kernels/BCs for solving the biharmonic equation (which requires C1 elements) and tests the Laplacian jump indicator with this PDE instead.

In the course of investigating this issue, it was discovered that the way MOOSE imposes Dirichlet boundary conditions is not correct for Hermite elements, since the constraints must involve not only the value dofs but the gradient dofs as well. The current workaround is to use penalty BCs for these tests, but there are known drawbacks to this. A longer term fix is to look into the `DirichletBoundary` objects in libMesh to see if they can be used in MOOSE, or to employ a non-penalty approach like Nitsche's method for imposing BCs (I know this works for H1 problems and Lagrange elements, not sure about the Biharmonic problem.)

Refs #2190.

### CODE REVIEW

**Comment by jwpeterson** [comment]
I also fixed the tests in `tests/variables/fe_hermite_convergence` since they had similar inconsistent weak forms, i.e. too many boundary conditions were imposed on each boundary.

File: test/include/bcs/BiharmonicLapBC.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * The weak form of the biharmonic equation has a term
+ * \int -Lap(u) * dv/dn ds
+ * which we use to weakly impose the value of Lap(u) on the boundary.
+ */
+class BiharmonicLapBC : public IntegratedBC
+{
+public:
+  BiharmonicLapBC(const InputParameters & parameters);
+  virtual ~BiharmonicLapBC() {}
```
Comment (permcody) [inline test/include/bcs/BiharmonicLapBC.h]: I usually don't bother putting these in on derived classes anymore.

File: test/include/bcs/BiharmonicLapBC.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+protected:
+  virtual Real computeQpResidual();
```
Comment (permcody) [inline test/include/bcs/BiharmonicLapBC.h]: override

**Review by permcody — CHANGES_REQUESTED** [review]
Very minor comments. Looks much better, thanks! I guess we don't require .md files for test objects?

**Comment by jwpeterson** [comment]
> I guess we don't require .md files for test objects?

Shhh don't tell Andrew...

**Comment by jwpeterson** [comment]
BTW, here's what the error indicator looks like as we refine the grid for the new problem (true solution is a Gaussian centered at the origin).

error_10x10
error_20x20
error_40x40
error_80x80

**Review by permcody — APPROVED** [review]
